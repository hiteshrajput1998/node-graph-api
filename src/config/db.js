import mongoose from 'mongoose';

export default async (db) => {
        try {
        const conn = await mongoose.connect('mongodb://127.0.0.1:27017/CollegeDB');

        // const conn = await mongoose.connect(`mongodb://${process.env.MONGOOSE_DB_HOST}:${process.env.MONGOOSE_DB_PORT}/${process.env.MONGOOSE_DB_NAME}`);
        return conn;
    } catch (err) {
        throw new Error(`MongoDB connection error: ${err}`);
    }
};