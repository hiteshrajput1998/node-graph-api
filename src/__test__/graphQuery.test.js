import { gql } from 'apollo-server';
import testClient from '../handler';
import db from '../config/db';

const { query, mutate } = testClient;

beforeEach(() => {
    db();
});

describe('GraphQuery CRUD', () => {

    it('getCollege', async () => {
        const GET_COLLEGES = gql`
            query{
                getcolleges{
                    id
                    name
                    address
                }
            }
        `;

        const device = {
            inputField: {
                name: "WW",
                address: "Ah'ed"
            }
        };

        const { data } = await query({
            mutation: GET_COLLEGES,
        });

        expect(data).toMatchSnapshot();
    });

    it('getColleges emptyObject', async () => {
        const GET_COLLEGES = gql`
            query{
                getcolleges{
                    id
                    name
                    address
                }
            }
        `;

        const { data } = await query({
            query: GET_COLLEGES
        });

        expect(data).toMatchObject({ "getcolleges": [] });
    });

    afterAll(async () => {
    });
});