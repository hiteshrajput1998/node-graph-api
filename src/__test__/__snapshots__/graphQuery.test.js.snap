// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQuery CRUD getCollege 1`] = `
Object {
  "getcolleges": Array [
    Object {
      "address": "UniversityArea",
      "id": "60489cc69deb2428b83cd28c",
      "name": "KS2",
    },
    Object {
      "address": "UniversityArea",
      "id": "6049ba2aab290b2ab87f893c",
      "name": "HRR",
    },
    Object {
      "address": "University",
      "id": "604a05d0daea693d703118fc",
      "name": "MS Uni",
    },
    Object {
      "address": "Maninagar",
      "id": "604b215ffabc4d56a06e408b",
      "name": "HRR",
    },
    Object {
      "address": "Navrangpura",
      "id": "604b237c439f464ea48d19e7",
      "name": "KS2",
    },
  ],
}
`;

exports[`GraphQuery CRUD getColleges emptyObject 1`] = `
Object {
  "getcolleges": Array [
    Object {
      "address": "UniversityArea",
      "id": "60489cc69deb2428b83cd28c",
      "name": "KS2",
    },
    Object {
      "address": "UniversityArea",
      "id": "6049ba2aab290b2ab87f893c",
      "name": "HRR",
    },
    Object {
      "address": "University",
      "id": "604a05d0daea693d703118fc",
      "name": "MS Uni",
    },
    Object {
      "address": "Maninagar",
      "id": "604b215ffabc4d56a06e408b",
      "name": "HRR",
    },
    Object {
      "address": "Navrangpura",
      "id": "604b237c439f464ea48d19e7",
      "name": "KS2",
    },
  ],
}
`;
